lib/APMSpan.js

```diff
@@ -149,7 +159,11 @@ class APMSpan {
         }
         this._finished = true;
         this._endMillis = finishTime;
-        this._tracer.getRecorder().record(this);
+
+        // only parent span should be recorded -> one Trace with all nodes
+        if (this._spanContext.getTrace().isParent(this)) {
+            this._tracer.getRecorder().record(this);
```

Comment:
We can't rely on the trace fragment being complete when the top level span finishes. Although async actions should be performed using a FOLLOWS_FROM reference, this may not be guaranteed - so may be better to use some form of reference counting of spans associated with the same trace.